app/scripts/controllers/transactions/index.js

```diff
@@ -202,8 +202,36 @@ export default class TransactionController extends EventEmitter {
   @emits ${txMeta.id}:unapproved
   */
   addTransaction(txMeta) {
+    const {
+      type,
+      status,
+      chainId: id,
+      origin: referrer,
+      metamaskNetworkId,
+      txParams: { gasPrice },
+    } = txMeta;
+
+    const networkId = this.networkStore.getState();
+    const network = metamaskNetworkId || networkId;
+    const chainId = id || this._getCurrentChainId();
```

Comment:
@brad-decker let me know if these are the incorrect fallbacks for both network and chainId

Reply:
If you just leave these undefined they'll be filled by the metametrics controller with the current values so you shouldn't need to get the current network/chainId here. :D

Reply:
LGTM though other than that.

Reply:
👍 will update per our call

Reply:
fixed